### PR TITLE
Fix available device key file path type

### LINKED ISSUE #3942: AvailableDevice.key_file_path returns incorrect type
Returns a `str` instead of a `Path`.

```python
>>> from parsec._parsec import list_available_devices
>>> type(list_available_devices("/tmp/parsec-testenv-v9lgjm12/config/parsec")[0].key_file_path)
<class 'str'>

```

### PR DESCRIPTION
**Pull request by Volham22**
# What has changed ?

This closes #3942 . When we return a `PathBuf` with `pyo3` the default conversion gives a python string. We want this conversion to yield a `pathlib.Path` because it makes path manipulation easier in python.

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: src/binding_utils.rs [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
+
+impl IntoPy<PyObject> for PathWrapper {
+    fn into_py(self, py: pyo3::Python<'_>) -> pyo3::PyObject {
+        // Pathlib is part of the standard library
+        let pathlib_module = py.import("pathlib").unwrap();
+        let path_ctor = pathlib_module.getattr("Path").unwrap();
+
+        path_ctor
+            .call1(PyTuple::new(py, [self.0.to_str().unwrap()]))
+            .unwrap()
```
Comment (FirelightFlagboy) [inline src/binding_utils.rs:27]: Use `expect`

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: src/binding_utils.rs (hunk 1 above)
Comment (Volham22) [reply src/binding_utils.rs]: Done.

File: src/binding_utils.rs [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+        let pathlib_module = py
+            .import("pathlib")
+            .expect("import `pathlib` module failed");
+        let path_ctor = pathlib_module
+            .getattr("Path")
+            .expect("can't get `Path` from `pathlib");
+
+        path_ctor
+            .call1(PyTuple::new(py, [self.0.to_str().unwrap()]))
+            .unwrap()
```
Comment (FirelightFlagboy) [inline src/binding_utils.rs:31]: Missing this unwrap

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: src/binding_utils.rs [hunk 3]
```diff
@@ -1,14 +1,38 @@
 // Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
 
 use pyo3::{
-    exceptions::PyNotImplementedError, pyclass::CompareOp, types::PyByteArray, types::PyBytes,
-    types::PyFrozenSet, FromPyObject, PyAny, PyResult,
+    exceptions::PyNotImplementedError,
+    pyclass::CompareOp,
+    types::PyByteArray,
+    types::PyFrozenSet,
+    types::{PyBytes, PyTuple},
```
Comment (TimeEngineer) [inline src/binding_utils.rs:8]: group import

File: src/binding_utils.rs [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
+        // Pathlib is part of the standard library
+        let pathlib_module = py
+            .import("pathlib")
+            .expect("import `pathlib` module failed");
+        let path_ctor = pathlib_module
+            .getattr("Path")
+            .expect("can't get `Path` from `pathlib");
+
+        path_ctor
+            .call1(PyTuple::new(py, [self.0.to_str().unwrap()]))
```
[the remark this answers is not in the document]
Comment (Volham22) [reply src/binding_utils.rs]: Yes it works.

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**